### PR TITLE
feat: accept custom IAM role names and permissions boundaries

### PR DESCRIPTION
**Pull request by cursedquail**
## Which problem is this PR solving?

Organizations whose SCPs required a permissions boundary on every IAM role,
or a specific naming convention for roles, could not use these modules at
all: `iam:CreateRole` was denied before anything else happened.

## Short description of the changes

Every IAM role these modules create accepts a `*_role_name` and a
`*_role_permissions_boundary`, both defaulting to null. Previously each role
was named `name_prefix = var.name`, with no way to override it and no way to
attach a boundary.

- A null name still resolves to `name_prefix = var.name`, so an existing
  deployment sees no diff.
- The two Lambda roles pass through to `terraform-aws-modules/lambda`,
  which already accepted both inputs.
- An IAM role's name cannot change in place, so setting a `*_role_name` on
  a deployment created without one replaces the role — and with it the
  Firehose delivery stream, subscription filter, metric stream, or Lambda
  that references it, which means a brief gap in telemetry delivery.
  Boundaries update in place. README.md and every variable description
  carry the warning.

USAGE.md had drifted from `versions.tf` since the provider v6 bump in #108;
regenerating it also required `lockfile: false` in .terraform-docs.yml, so
the Providers table reports the `required_providers` constraint rather than
whatever version the regenerator happens to have locked locally.

## How to verify that this has the expected result

`tests/` instantiates each module at least once with both inputs set and at
least once with them null, so the CI apply covers both paths. The boundary
is a deliberately permissive `aws_iam_policy` — the test covers that a
boundary attaches, not what a boundary does.

### CODE REVIEW

**Review by vreynolds — APPROVED** [review]

File: USAGE.md [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
 | <a name="requirement_terraform"></a> [terraform](#requirement\_terraform) | >= 1.0 |
-| <a name="requirement_aws"></a> [aws](#requirement\_aws) | >= 5.0 |
+| <a name="requirement_aws"></a> [aws](#requirement\_aws) | ~> 6.0 |
 
 ## Providers
 
 | Name | Version |
 |------|---------|
-| <a name="provider_aws"></a> [aws](#provider\_aws) | >= 5.0 |
+| <a name="provider_aws"></a> [aws](#provider\_aws) | 6.57.1 |
```
Comment (robbkidd) [inline USAGE.md]: This looks like a number from your local terraforming. Should it be `~> 6.0`?

File: variables.tf [hunk 2]
```diff
@@ -195,18 +195,102 @@ variable "cloudwatch_logs_name" {
...
+  default     = null
+}
+
+variable "rds_logs_lambda_role_permissions_boundary" {
+  type        = string
+  description = "ARN of the permissions boundary policy to attach to the RDS log transform Lambda IAM role."
+  default     = null
+}
+
+variable "rds_logs_cloudwatch_role_name" {
```
Comment (robbkidd) [inline variables.tf]: For parity with the `cloudwatch_logs_role_name` that uses this variable?

```suggestion
variable "rds_logs_cloudwatch_logs_role_name" {
```

File: variables.tf [hunk 3]
```diff
@@ -195,18 +195,102 @@ variable "cloudwatch_logs_name" {
...
+  type        = string
+  description = "Name for the IAM role used by CloudWatch Logs in the RDS Logs integration. If null, Terraform generates a name using rds_logs_name as a prefix."
+  default     = null
+}
+
+variable "rds_logs_cloudwatch_role_permissions_boundary" {
```
Comment (robbkidd) [inline variables.tf]: Same parity with attribute it's used on.

```suggestion
variable "rds_logs_cloudwatch_logs_role_permissions_boundary" {
```

File: main.tf [hunk 4]
```diff
@@ -49,6 +54,13 @@ module "rds_logs" {
 
   count = var.enable_rds_logs ? 1 : 0
 
+  lambda_role_name                          = var.rds_logs_lambda_role_name
+  lambda_role_permissions_boundary          = var.rds_logs_lambda_role_permissions_boundary
+  cloudwatch_logs_role_name                 = var.rds_logs_cloudwatch_role_name
+  cloudwatch_logs_role_permissions_boundary = var.rds_logs_cloudwatch_role_permissions_boundary
```
Comment (robbkidd) [inline main.tf]: 🍐🇮🇹🍵  

```suggestion
  cloudwatch_logs_role_name                 = var.rds_logs_cloudwatch_logs_role_name
  cloudwatch_logs_role_permissions_boundary = var.rds_logs_cloudwatch_logs_role_permissions_boundary
```

**Review by robbkidd — CHANGES_REQUESTED** [review]
Some inline questions/suggested changes.

**Tests:** Can you add a test that exercises the non-null path? CI does an apply of what's in `tests/`. Happily, the `null` existing behavior there confirms this ought to be backwards compatible. 

**Warn folks who opt-in?:** Switching an existing deployment from generated-prefix to explicit name will force IAM role replacement which I think cascades into firehose/metric-stream referencen and Lambda roles. I personally don't run Terraform enough to know how much warning folks should have about that. Are there words that can be written in description or is that overkill?

File: USAGE.md (hunk 1 above)
Comment (robbkidd) [reply USAGE.md]: @cursedquail How about this fella?

Comment (cursedquail) [reply USAGE.md]: workin on it :)

gonna push a commit that'll fix this one but probably not the rest, and then uhhh I will sign off bc it's 5pm

Comment (robbkidd) [reply USAGE.md]: > gonna push a commit that'll fix this one

🎊 🥳 

> I will sign off bc it's 5pm

Same. We'll see what comes out of the CI oven in the morning!

Comment (cursedquail) [reply USAGE.md]: Had claude poke at it. It comes from our `make generate-docs` command, which, _if_ it finds a `.terraform.lock.hcl` file, will by default use the version in the lockfile. So I had run `terraform init` to validate the changes, and then a later `make generate-docs` picked this up.

Claude hath suggested adding `lockfile: false` to this file https://github.com/honeycombio/terraform-aws-integrations/blob/0bdbd9bae48eaabe90fa5e767b095268ea314d65/.terraform-docs.yml and I'm gonna add that change to this PR

Comment (robbkidd) [reply USAGE.md]: Who are we to argue with Clod?

**Review by robbkidd — APPROVED** [review]
:shipit: 🚀 🍰

File: tests/rds.tf [hunk 5]
```diff
@@ -24,6 +24,32 @@ module "rds_mysql_logs" {
   s3_failure_bucket_arn = module.firehose_failure_bucket.s3_bucket_arn
 }
 
+// Same database, a different log type, and every IAM role named explicitly
```
Comment (robbkidd) [inline tests/rds.tf:27]: > 🤖 **Nice touch** on `rds_mysql_error_logs:` reuses the same RDS instance with `db_log_types = ["error"]` instead of standing up a second database. Different log group, so no subscription-filter collision, and no extra CI minutes.

give yourself a medal
